Source/WebCore/rendering/HitTestResult.cpp

```diff
@@ -219,6 +220,40 @@ bool HitTestResult::isSelected() const
     return frame->selection().contains(m_hitTestLocation.point());
 }
 
+bool HitTestResult::allowsFollowingLink() const
+{
+    auto linkURL = absoluteLinkURL();
+    if (linkURL.isEmpty())
+        return false;
+
+    RefPtr innerFrame = innerNodeFrame();
+    if (!innerFrame)
+        return false;
+
+    RefPtr document = innerFrame->protectedDocument();
```

Comment:
`RefPtr document = innerFrame->document();`

We do not call `protectedX()` variants to initialize variables. The name is longer and there is no benefit.